osutil/cp_test.go

```diff
@@ -241,3 +242,89 @@ func (s *cpSuite) TestCopyPreserveAll(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(st1.ModTime(), Equals, st2.ModTime())
 }
+
+func (s *cpSuite) TestCopyPreserveAllSync(c *C) {
+	dir := c.MkDir()
+	cp := filepath.Join(dir, "cp")
+	sync := filepath.Join(dir, "sync")
+	log := filepath.Join(dir, "log")
+
+	sh := fmt.Sprintf(`#!/bin/sh
+echo "$0:$@" >> %s
+`, log)
+
+	c.Assert(ioutil.WriteFile(cp, []byte(sh), 0755), IsNil)
+	c.Assert(os.Symlink(cp, sync), IsNil)
+
+	oldPATH := os.Getenv("PATH")
+	defer os.Setenv("PATH", oldPATH)
+	os.Setenv("PATH", dir+":"+oldPATH)
+
+	src := filepath.Join(dir, "meep")
+	dst := filepath.Join(dir, "copied-meep")
+
+	err := ioutil.WriteFile(src, []byte(nil), 0644)
+	c.Assert(err, IsNil)
+
+	err = CopyFile(src, dst, CopyFlagPreserveAll|CopyFlagSync)
+	c.Assert(err, IsNil)
+
+	content, err := ioutil.ReadFile(log)
+	c.Assert(err, IsNil)
+	c.Check(string(content), Equals, fmt.Sprintf("%[1]s/cp:-av %[1]s/meep %[1]s/copied-meep\n%[1]s/sync:\n", dir))
+}
+
+func (s *cpSuite) TestCopyPreserveAllSyncCpFailure(c *C) {
+	dir := c.MkDir()
+	cp := filepath.Join(dir, "cp")
```

Comment:
because i didn't know it and just repeated what's done elsewhere. I'll look at MockCmd and use it.